c7n/resources/asg.py

```diff
@@ -492,6 +492,53 @@ def process(self, asgs, event=None):
         return super(VpcIdFilter, self).process(asgs)
 
 
+@filters.register('launch-failure')
```

Comment:
this should be generic filter for asg activity, not hard-coded to launch failures.